uavcan/diagnostic/65520.Text.1.0.uavcan

```diff
@@ -0,0 +1,15 @@
+#
+# Generic human-readable text message for logging and displaying purposes.
+# Generally it should be published at the lowest priority level.
+#
+
+# Standard severity, 2 bit wide.
+Severity.1 severity
+
+# Short name of the message source within the publishing node.
+void1
+uint8[<32] source
+
+# The actual useful text of the message.
+# Normally, messages should be kept as short as possible, especially those of high severity.
+uint8[<=200] text
```

Comment:
We have to make our definitions compatible with resource-constrained nodes. That limits the maximum size of serialized messages -- I briefly mentioned that in the README, saying that the size should not be larger than some rather arbitrarily chosen limit. Now, messages and services differ in the amount of context that is maintained around them in the application when they are processed by the protocol stack. Simple services are cheap memory-wise, because usually at the time of responding to a service call the execution flow is situated somewhere in a callback from the protocol stack where the amount of contextual information is usually low, and as a result, the required stack depth is also low, which results in lower worst-case memory footprint. Certain other definitions, such as this message, for example, may be utilized in context-rich scenarios, e.g. if a log message is emitted from a sophisticated control loop, and so on, where the stack utilization can be high. To simplify application design and not force the designer to resort to various countermeasures, we need to account for that, and limit the worst case memory size of messages that are expected to affect worst-case stack utilization.

Other than that, there is no strict justification for 200 bytes exactly.

Reply:
I don't think a lot of resource-constrained nodes will read diagnostics. For sending they can just use a lower bound internally.

On the other hand, the old definition was <= 90. Is there any reason to increase this number?

Reply:
> On the other hand, the old definition was <= 90. Is there any reason to increase this number?

We don't have enough evidence to come up with a rigorously defined value here. I looked through some of the code I've written and decided that it might be useful to raise the limit, which I did. We can easily reduce it back or increase further (mind the above) if it is considered beneficial.

Reply:
I don't like changing from one arbitrary value to a different arbitrary value. My proposal would be to either just use the whole uint8 for indexing and define it as [<=255]. This is the "we don't know so we just give as much leeway as possible" approach. The number will then be rooted in the fact that a longer dynamic array would need another byte for its length, and that it's very reasonable to think that 255 is long enough.

The alternative approach would delay stabilizing until we can gather data on how long these strings normally are and set a value based on this.

Since resource constrained nodes will not read log/diagnostics most of the time, the 255 bytes approach should be fine.

Reply:
Increased to 255 max.